lib/pacemaker/pcmk_sched_messages.c

```diff
@@ -36,13 +36,17 @@ log_resource_details(pe_working_set_t *data_set)
      */
     all = g_list_prepend(all, strdup("*"));
 
-    out = pcmk__new_logger();
-    if (out == NULL) {
-        return;
-    }
+    if (!pcmk__str_eq(((pcmk__output_t *) data_set->priv)->fmt_name,
+                      "log", pcmk__str_none)) {
+        logger_out = pcmk__new_logger();
+        if (logger_out == NULL) {
+            return;
+        }
```

Comment:
Nothing wrong with this approach, but I noticed crm_simulate directly calls LogNodeActions() when not quiet. If we're lucky, maybe we can just use the data_set output object here, and get rid of the direct call in crm_simulate.

Reply:
Making these changes has the following side effects:

* .scores files now have node and resource information written to them, like so:
```
--- a/cts/scheduler/scores/1-a-then-bm-move-b.scores
+++ b/cts/scheduler/scores/1-a-then-bm-move-b.scores
@@ -1,4 +1,6 @@
 
+A      (ocf:heartbeat:Dummy):   Started 18node1
+B      (ocf:heartbeat:Dummy):   Started 18node2
 pcmk__native_allocate: A allocation score on 18node1: 0
 pcmk__native_allocate: A allocation score on 18node2: 0
 pcmk__native_allocate: A allocation score on 18node3: 0
```

* .summary files have fencing messages removed and spacing has changed elsewhere, like so:
```
--- a/cts/scheduler/summary/year-2038.summary
+++ b/cts/scheduler/summary/year-2038.summary
@@ -46,11 +46,10 @@ Current cluster status:
       * openstack-cinder-volume-docker-0       (ocf:heartbeat:docker):  Started controller-0
 
 Transition Summary:
-  * Fence (reboot) overcloud-novacompute-1 'remote connection is unrecoverable'
-  * Stop       overcloud-novacompute-1              (            controller-1 )   due to node availability
+  * Stop       overcloud-novacompute-1              (            controller-1 )  due to node availability
   * Start      ip-10.0.0.110                        (            controller-1 )
   * Recover    stonith-fence_compute-fence-nova     (            controller-2 )
-  * Stop       compute-unfence-trigger:1            ( overcloud-novacompute-1 )   due to node availability
+  * Stop       compute-unfence-trigger:1            ( overcloud-novacompute-1 )  due to node availability
```

These are differences from my patch updating the output of cts/cts-scheduler, NOT from master.  So, I am wondering if those fencing messages ever should have been there.  The whitespace I'm not concerned about.  And then the node and resource stuff in the .scores files... that might point to a bug elsewhere.  I have not checked how this would look on a running system.  That's the next step.

Reply:
The fencing messages coem from that call to LogNodeActions in crm_simulate.c, around line 1073.  So I don't see how we could get rid of that.

Reply:
And then, the resources in the .scores file come from the call to log_resource_details after stage0 in pcmk__schedule_actions.  That can be taken care of by just adding a check for pe_flag_print not being set, since I think we only want to do that logging if we're running a daemon.  There's no point in logging if crm_simulate is being run.  So, that will at least let me get rid of the new logger setup in log_resource_details.

Reply:
There's a pcmk__is_daemon global boolean

Reply:
I refuse to be excited about the existence of a global variable, but I think it's going to be useful here.  I think it can probably eliminate the new pe_flag_print flag that I added.